### PR TITLE
Force arch when running in a bun postinstall process

### PR DESCRIPTION
**Pull request by elmpp**
Fixes problem when installing via Bun package manager.

Bun runs its package manager process with the correct process.arch var. This env var influences Moon's arch-specific core dependency.

Bun runs postinstall processes however via node. Node does not have, or does not have set, the same process.arch which results in the postinstall process resolving the wrong core dependency

Bun thankfully sets a useful value at process.npm_config_user_agent which allows a safe edge case solution here

### CODE REVIEW

File: .yarn/versions/baeeda33.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+releases:
+  "@moonrepo/cli": patch
```
Comment (milesj) [inline .yarn/versions/baeeda33.yml:2]: need to bump all the core packages too, try `yarn version:bump:bin patch`

Comment (elmpp) [reply .yarn/versions/baeeda33.yml:2]: 👍

File: packages/cli/postinstall.js [hunk 2]
```diff
@@ -13,7 +13,8 @@ const isMacos = process.platform === 'darwin';
 const isWindows = process.platform === 'win32';
 
 const platform = isWindows ? 'windows' : isMacos ? 'macos' : process.platform;
-const parts = [platform, process.arch];
+const arch = process.env['npm_config_user_agent'].match(/^bun.*arm64$/) ? 'arm64' : process.arch; // https://github.com/moonrepo/moon/issues/1103
```
Comment (milesj) [inline packages/cli/postinstall.js:16]: @elmpp Is `npm_config_user_agent` always set, even for npm/pnpm/yarn? Maybe we need a property existence check?

Comment (elmpp) [reply packages/cli/postinstall.js]: Added an old-style conditional 👍

**Comment by milesj** [comment]
awesome thanks!

**Comment by elmpp** [comment]
Thank you 🙏

**Pull request merged**